fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -196,7 +248,19 @@ private void incrementCallCount(FDBStoreTimer.Counts counter) {
     public void writeFDBLuceneFileReference(@Nonnull final String name, @Nonnull final FDBLuceneFileReference reference) {
         LOGGER.trace("writeFDBLuceneFileReference {}", reference);
         incrementCallCount(FDBStoreTimer.Counts.LUCENE_WRITE_FILE_REFERENCE);
-        context.ensureActive().set(metaSubspace.pack(name), reference.getTuple().pack());
+        String mergedFileName = getMergedFileName(name);
+        if (mergedFileName != null) {
+            FDBLuceneMiltiFileReference multiFileReference = context.asyncToSync(FDBStoreTimer.Waits.WAIT_LUCENE_FILE_LENGTH, getMultiFDBLuceneFileReference(mergedFileName));
```

Comment:
My understanding of the motivation for this PR is to reduce the read requests to FoundationDB server for file references for compounded index. Looks one side effect is to have read-and-write instead of only one write when we want to write the reference for the 3 files to FoundationDB. Was there any discussion or consideration on the tradeoff between the less reads and potentially more expensive writes of the references? This comment is not to request any changes, just to understand more context about this optimization.